multinet/user.py

```diff
@@ -0,0 +1,98 @@
+"""User data and functions."""
+
+from uuid import uuid4
+from arango.collection import StandardCollection
+
+from multinet.db import db
+from multinet.auth.types import GoogleUserInfo, UserInfo, User
+
+from typing import Optional, Dict
+
+
+# This essentially duplicated the UserInfo TypedDict.
+# Is this a good reason to switch to dataclasses?
+USER_FIELDS = {"email", "name", "family_name", "given_name", "picture", "sub"}
```

Comment:
Isn't there a way to extract the fields from `UserInfo`?

Reply:
There doesn't seem to be, as at runtime, any `TypedDict` is just a regular `dict`.

Reply:
If you make `UserInfo` a dataclass, then I wonder if you could call [`.fields()`](https://docs.python.org/3/library/dataclasses.html) on the class itself and get this information.

Reply:
That's the main reason I proposed this. I'm currently giving this a shot to see what it looks like.

Reply:
Addressed by 15d5819.